src/Features/Core/Portable/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -516,15 +668,26 @@ private int NonParamsParameterCount(IMethodSymbol method)
                 return type.IsReferenceType || type.IsNullable();
             }
 
+            if (type.Kind == SymbolKind.TypeParameter)
+            {
+                return true;
+            }
+
             return false;
         }
 
-        private class MyCodeAction : CodeAction.DocumentChangeAction
+        private class MyCodeAction : CodeAction.SolutionChangeAction
+        {
+            public MyCodeAction(string title, Func<CancellationToken, Task<Solution>> createChangedSolution)
+                : base(title, createChangedSolution)
+            {
+            }
+        }
+
+        private class GroupingCodeAction : CodeAction.CodeActionWithNestedActions
```

Comment:
don't need this.  you can just directly instantiate "CodeActionWithNestedActions"